File: docs/petsc/helloworld.c
//  First try to print multiple "Hello, World !" on the terminal with the
//  PETSc library.
//  Julien VALENTIN (July, 2022)  julien.vlnt@gmail.com
//  usage : `make hello && $PETSC_DIR/lib/petsc/bin/petscmpiexec -n 8 ./hello`
#include <petscsys.h>

int main(int argc, char ** argv)
{
    PetscMPIInt rank;
    
    PetscInitialize(&argc, &argv, NULL, NULL);

    MPI_Comm_rank(PETSC_COMM_WORLD, &rank);

    PetscSynchronizedPrintf(PETSC_COMM_WORLD, "Rank %d says hello !\n", rank);
    PetscSynchronizedFlush(PETSC_COMM_WORLD, PETSC_STDOUT);

    PetscFinalize();

    return 0;
}

File: docs/petsc/helloworld.f90
#include "include/finclude/petsc.h"

program hello

    integer ierr, rank    ! useful integers

    ! Start 
    call PetscInitialize(PETSC_NULL_CHARACTER, ierr)
    
    ! Read from `com_world` the rank of the current job
    call MPI_Comm_rank(PETSC_COMM_WORLD, rank, ierr)
    
    ! Print hello, one per job
    call PetscSynchronizedPrintf(PETSC_COMM_WORLD, "Rank %d says hello !\n", rank);
    
    ! Synchronization of printing
    call PetscSynchronizedFlush(PETSC_COMM_WORLD, PETSC_STDOUT);
    
    ! End of the program
    call PetscFinalize(ierr)
end program
